### PR TITLE
operator/network: add OpenShiftSRIOV network type

### PR DESCRIPTION
**Pull request by zshi-redhat**
This API change is part of [openshift/cluster-network-operator#84](https://github.com/openshift/cluster-network-operator/pull/84).

### CODE REVIEW

**Comment by zshi-redhat** [comment]
@dcbw @knobunc PTAL.

**Comment by zshi-redhat** [comment]
@danwinship please help to take a look at this API change of cluster network operator that adds OpenShiftSRIOV supports.

File: operator/v1/types_network.go [hunk 1]
```diff
@@ -111,6 +111,10 @@ type AdditionalNetworkDefinition struct {
 	// rawCNIConfig is the raw CNI configuration json to create in the
 	// NetworkAttachmentDefinition CRD
 	RawCNIConfig string `json:"rawCNIConfig"`
+
+	// sriovConfig configures the openshiftSRIOV plugin.
+	// +optional
+	SRIOVConfig *SRIOVConfig `json:"sriovConfig,omitempty"`
```
Comment (danwinship) [inline operator/v1/types_network.go]: should mark `RawCNIConfig` as `omitempty` now too

Comment (zshi-redhat) [reply operator/v1/types_network.go]: Thanks Dan! I noticed that this has been changed in Tomo's [PR](https://github.com/openshift/api/pull/257), so I'll keep as it is for this PR, does it sound ok?

**Comment by zshi-redhat** [comment]
@adambkaplan @abhinavdahiya PTAL?

**Comment by danwinship** [comment]
/hold
not for 4.1 now

**Comment by zshi-redhat** [comment]
Closing per https://github.com/openshift/cluster-network-operator/pull/206

**Pull request closed without merging**